### ISSUE TITLE
Osquery port to Powerpc

### ISSUE THREAD

**Issue by ginsburgnm** [osquery/osquery#7067]
# Feature request

### What new feature do you want?

I have successfully compiled osquery on ppc64le, I was wondering if this team would be open to merging code into this repo to support power.

### How is this new feature useful?

Adds support for a new architecture, this architecture is used in multiple HPC environments.

### How can this be implemented?

Upgrade (or patch) required for glibc -> there is an assembly code patch required on this file `crosstool-ng/.build/src/glibc-2.23/sysdeps/powerpc/powerpc64/power6/memset.S` later versions of the library have patched this.

changes required to following files (I just copied and pasted my `git status`):
```
        modified:   CMakeLists.txt
        modified:   cmake/globals.cmake
        modified:   libraries/cmake/source/aws-sdk-cpp/CMakeLists.txt
        modified:   libraries/cmake/source/aws-sdk-cpp/aws-checksums_src (modified content)
        modified:   libraries/cmake/source/boost/CMakeLists.txt
        modified:   libraries/cmake/source/ebpfpub/src (modified content)
        modified:   libraries/cmake/source/libgcrypt/CMakeLists.txt
        modified:   libraries/cmake/source/libgcrypt/src (modified content)

```
New files are required within the following directories
```

        libraries/cmake/source/libaudit/config/ppc64le/
        libraries/cmake/source/libdpkg/config/ppc64le/
        libraries/cmake/source/libgcrypt/config/ppc64le/
        libraries/cmake/source/libgcrypt/generated/ppc64le/
        libraries/cmake/source/libgpg-error/config/ppc64le/
        libraries/cmake/source/libgpg-error/generated/ppc64le/
        libraries/cmake/source/librdkafka/config/ppc64le/
        libraries/cmake/source/lldpd/config/ppc64le/
        libraries/cmake/source/lzma/config/ppc64le/
        libraries/cmake/source/smartmontools/config/ppc64le/
```

So as we can see, some upstream changes need to be made as well to OsQuery's dependencies.

### Considerations

*potential problem* port to power seems to require a later version of cmake than osquery wants to utilize, I was not able to build the `osquery-toolchain` with 3.17.5 on ppc64le, I utilized 3.20.1; it may be that a version in between the two would work. I have not extensively tested. However, minimal changes (other than running a newer version of cmake) were required to build the toolchain (Changes for toolchain were in 3 files: `build.sh` `config` and the addition of a ` crosstool-ng-config-ppc64le`)

I also understand if your team would not want to merge my changes in due to not having powerpc machines available for purposes of things such as CI or future maintenance; they are not particularly common in data centers outside of HPC.

Further note, my changes are not immediately ready; I did some hacky stuff as POC, so I need to do cleanup.

Please let me know if more information is required, thanks!

**Comment by theopolis** [comment]
This is cool work! Jumping right to the chase, I don't think we can support this officially unless there's a reasonable solution for CI. We are currently halfway through finding official support for aarch64 and it's requiring quite the investment for testing.

We can support making small build changes here and there if you submit PRs to make building on ppc64le easier.

**Comment by ginsburgnm** [comment]
Thanks for the response! I completely understand, I'll look into some small non-obtrusive build changes.

**Comment by milind69** [comment]
Hi @ginsburgnm 
I am probably facing same issues , where did you get file for ppc64le ? 
i copied from x86_64 

 1222  cp -rp ./libaudit/config/x86_64 ./libaudit/config/ppc64le
 1229  cp -rp ./libdpkg/config/x86_64 ./libdpkg/config/ppc64le
 1230  cp -rp ./libgcrypt/config/x86_64 ./libgcrypt/config/ppc64le
 1231  cp -rp ./libgcrypt/generated/x86_64 ./libgcrypt/generated/ppc64le
 1232  cp -rp ./libgpg-error/config/x86_64 ./libgpg-error/config/ppc64le
 1233  cp -rp ./libgpg-error/generated/x86_64 ./libgpg-error/generated/ppc64le
 1234  cp -rp ./librdkafka/config/x86_64 ./librdkafka/config/ppc64le
 1235  cp -rp ./lldpd/config/x86_64 ./lldpd/config/ppc64le
 1236  cp -rp ./lzma/config/x86_64 ./lzma/config/ppc64le
 1237  cp -rp ./smartmontools/config/x86_64 ./smartmontools/config/ppc64le

but my compile fails 

home/u347729/opt/osquery-toolchain/final/sysroot/usr/lib/clang/11.0.0/include/altivec.h:238:53: error: unknown type name 'vector'
                                                    vector bool short __b) {
                                                    ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
gmake[2]: *** [libs/src/lzma/CMakeFiles/thirdparty_lzma.dir/build.make:772: libs/src/lzma/CMakeFiles/thirdparty_lzma.dir/src/src/liblzma/lz/lz_encoder.c.o] Error 1
gmake[2]: Leaving directory '/home/dciunix/toolchain/osquery/build'
gmake[1]: *** [CMakeFiles/Makefile2:5387: libs/src/lzma/CMakeFiles/thirdparty_lzma.dir/all] Error 2
gmake[1]: Leaving directory '/home/dciunix/toolchain/osquery/build'
gmake: *** [Makefile:150: all] Error 2


I am compiling on ppc64le ..
how did you compile ? where can I find your compile instruction ?

**Comment by directionless** [comment]
> Thanks for the response! I completely understand, I'll look into some small non-obtrusive build changes.

Without seeing the changes, much of what you quoted seems small. Of course, it depends on the details.

**Comment by ginsburgnm** [comment]
@milind69 Yeah, you can't just copy the configs, they need to be modified for ppc. I got buried down with other things, I'll try to dig my configs up for you in the next couple days.
